### PR TITLE
[New Rule] SSH Authorized Keys File Modified Inside a Container

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/2776

### Description
This rule detects the creation or modification of an authorized_keys or sshd_config file inside a container. 
The Secure Shell (SSH) authorized_keys file specifies which users are allowed to log into a server using public key authentication. 
Adversaries may modify it to maintain persistence on a victim host by adding their own public key(s). 
Unexpected and unauthorized SSH usage inside a container can be an indicator of compromise and should be investigated. 

### Query
```
file where container.id:"*" and
  event.type in ("change", "creation") and file.name: ("authorized_keys", "authorized_keys2", "sshd_config")
```
<details>
<summary/> Updated query to include sshd_config and remove file paths </summary>

image
</details>

### CODE REVIEW

File: rules/integrations/cloud_defend/persistence_ssh_authorized_keys_modification_inside_a_container.toml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+rule_id = "f7769104-e8f9-4931-94a2-68fc04eadec3"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Lateral Movement", "Container"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where container.id:"*" and
+  event.type in ("change", "creation") and 
+  event.action in ("creation", "modification") and
```
Comment (DefSecSentinel) [inline rules/integrations/cloud_defend/persistence_ssh_authorized_keys_modification_inside_a_container.toml:33]: I'm not sure you need both event.type and event.action here. It's redundant as they both specify the same activity. I think could choose one or the other and get the same result.

**Review by Aegrah — APPROVED** [review]
LGTM!

File: rules/integrations/cloud_defend/persistence_ssh_authorized_keys_modification_inside_a_container.toml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Lateral Movement", "Container"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where container.id:"*" and
+  event.type in ("change", "creation") and 
+  event.action in ("creation", "modification") and 
+  file.name: ("authorized_keys", "authorized_keys2", "sshd_config")
```
Comment (Samirbous) [inline rules/integrations/cloud_defend/persistence_ssh_authorized_keys_modification_inside_a_container.toml:34]:
```suggestion
file where container.id:"*" and
  event.type in ("change", "creation") and file.name: ("authorized_keys", "authorized_keys2", "sshd_config")
```

**Review by Samirbous — APPROVED** [review]

**Comment by Danouchka** [comment]
Hi 
I have the following issue with that rule in 8.17.1 , latest security rules version. 
Capture d’écran 2025-02-03 à 21 58 06

**Pull request merged**